include/sys/vm_map.h

```diff
@@ -9,6 +9,11 @@
 typedef struct vm_map vm_map_t;
 typedef struct vm_segment vm_segment_t;
 
+typedef enum {
+  VM_SEG_PRIVATE = 0, /* private memory (default) */
+  VM_SEG_SHARED = 1,  /* shared memory */
+} vm_seg_flags_t;
+
```

Comment:
I don't think we need `VM_SEG_PRIVATE`, since we need only a single bit to convey the private/shared status of a segment: if `VM_SEG_SHARED` is set, the segment is shared, and if it's not set, it is private. Also, having an enum member with value 0 for a flag type is weird (how do we test for it)? I therefore suggest removing `VM_SEG_PRIVATE`.

Reply:
I assumed that maybe in the future we would like to check if a given segment is private or shared. And if you look at vm_flags_t enum, you could see the similar behavior there: there is a default VM_FILE flag, the value for this member is equal to zero. So similar question would be here: why wouldn't we remove this flag too? I think it's important to keep consistency in code and use the same rules.

Reply:
`vm_flags_t` is supposed to reflect the `MAP_*` flags, so `VM_FILE` is 0 because `MAP_FILE` is 0, and we can't change `MAP_FILE` because it's exposed to userspace and it's the default value. Besides, having a 0 valued flag enum member is dangerous when a user wants to check a flag: when I want to check if a segment is shared, do I check `(flags & VM_SEG_SHARED)` or `!(flags & VM_SEG_PRIVATE)`? Clearly the latter is wrong, but it's not obvious at all without reading the enum definition.

Reply:
Done.